engine/src/main/java/graphql/nadel/engine/execution/ServiceResultNodesToOverallResult.java

```diff
@@ -594,12 +590,46 @@ public TraversalControl enter(TraverserContext<ExecutionResultNode> context) {
         });
     }
 
-    private List<String> getFieldIdsWithoutTransformationId(ExecutionResultNode node, TransformationMetadata transformationMetadata) {
-        return node.getFieldIds().stream().filter(fieldId -> FieldMetadataUtil.getTransformationIds(fieldId, transformationMetadata.getMetadataByFieldId()).size() == 0).collect(Collectors.toList());
+    private ExecutionResultNode nodesWithoutTransformationIds(ExecutionResultNode executionResultNode, Set<String> transformationIds, TransformationMetadata transformationMetadata) {
```

Comment:
Just to check that I understand what this method is doing: 
It will return a copy of the `executionResultNode` parameter, which I'm assuming is the root node, containing only fields that weren't transformed. And sub-nodes that ended up with no fields will be deleted?

Reply:
> containing only fields that weren't transformed
containing only fields that don't have any root transformations for them
if all fields have a root transformation for them, then yes, the field is deleted from that "copy"